### PR TITLE
Termux fixes

### PR DESCRIPTION
**Pull request by Sonicadvance1**

### CODE REVIEW

File: External/FEXCore/Source/Utils/Allocator/IntrusiveArenaAllocator.h [hunk 1]
```diff
@@ -7,14 +7,21 @@
 
 #include <bitset>
 #include <cstddef>
+#ifdef TERMUX_BUILD
+#include <experimental/memory_resource>
+#include <experimental/list>
+namespace fex_pmr = std::experimental::pmr;
```
Comment (neobrain) [inline External/FEXCore/Source/Utils/Allocator/IntrusiveArenaAllocator.h:13]: How about adding a feature test to see when we can remove this workaround?
```cpp
#if __has_include(<memory_resource>)
#error "Workaround can be removed"
#endif
```

Comment (Sonicadvance1) [reply External/FEXCore/Source/Utils/Allocator/IntrusiveArenaAllocator.h:18]: Resolved.

File: Source/Tests/CMakeLists.txt [hunk 2]
```diff
@@ -2,52 +2,61 @@ add_subdirectory(LinuxSyscalls)
...
-)
-
-if (CMAKE_BUILD_TYPE MATCHES "RELEASE")
-  target_link_options(FEXLoader
-    PRIVATE
-      "LINKER:--gc-sections"
-      "LINKER:--strip-all"
-      "LINKER:--as-needed"
-  )
+function(BuildFEX Name)
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply Source/Tests/CMakeLists.txt]: Removed the cmake function and simplified by just changing the `ln -s` to a `cp` in the termux specific workaround.
Resolved.

File: Source/Tools/CMakeLists.txt [hunk 3]
```diff
@@ -2,10 +2,13 @@ if (ENABLE_VISUAL_DEBUGGER)
   add_subdirectory(Debugger/)
 endif()
 
-add_subdirectory(FEXConfig/)
+if (NOT TERMUX_BUILD)
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply Source/Tools/CMakeLists.txt]: Removed this commit. X11 concern was removed due to a different package fixing the problem. Resolved.

File: Source/Tests/LinuxSyscalls/x64/Thread.cpp [hunk 4]
```diff
@@ -67,8 +67,13 @@ namespace FEX::HLE::x64 {
     REGISTER_SYSCALL_IMPL_X64(set_robust_list, [](FEXCore::Core::CpuStateFrame *Frame, struct robust_list_head *head, size_t len) -> uint64_t {
       auto Thread = Frame->Thread;
       Thread->ThreadManager.robust_list_head = reinterpret_cast<uint64_t>(head);
+#ifdef TERMUX_BUILD
+      // Termux doesn't support `set_robust_list` will get caught by seccomp
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply Source/Tests/LinuxSyscalls/x64/Thread.cpp]: Updated comment to be more specific about what is happening here. Resolved.

**Review by neobrain — APPROVED** [review]

**Pull request merged**